packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -78,20 +80,19 @@ export type MetadataAction =
     | ReturnType<typeof setAccountAdd>;
 
 // needs to be declared here in top level context because it's not recommended to keep classes instances in redux state (serialization)
-const providerInstance: Record<
-    DataType,
-    DropboxProvider | GoogleProvider | FileSystemProvider | undefined
-> = {
+const providerInstance: Record<DataType, AbstractMetadataProvider | undefined> = {
     labels: undefined,
 };
+
 const fetchIntervals: { [deviceState: string]: any } = {}; // any because of native at the moment, otherwise number | undefined
 
 const createProviderInstance = (
     type: MetadataProvider['type'],
     tokens: Tokens = {},
     environment: OAuthServerEnvironment = 'production',
     clientId?: string,
-) => {
+): DropboxProvider | GoogleProvider | FileSystemProvider | InMemoryTestProvider => {
```

Comment:
This needs to be exhaustive because `clientId`  property is not part of the parent AbstractClass. 

Shall be refactored in the future probably.